src/main/java/it/gov/pagopa/pu/debtpositions/service/InstallmentServiceImpl.java

```diff
@@ -73,4 +77,52 @@ public String updateInstallmentNotificationDate(UpdateInstallmentNotificationDat
     return debtPositionUpdateInstallmentService.updateInstallment(debtPositionDTO, updatedInstallments, wfExecutionParameters, accessToken, operatorExternalUserId).getRight();
   }
 
+  @Override
+  public InstallmentDTO updateInstallmentNotificationFee(Long organizationId, String nav, List<DebtPositionOrigin> debtPositionOrigin, long notificationFeeCents) {
+    List<InstallmentDTO> installments = installmentPIIRepository.getByOrganizationIdAndNav(organizationId, nav, debtPositionOrigin).stream()
+      .filter(installment ->
+        installment.getStatus().equals(InstallmentStatus.UNPAID) ||
+        installment.getStatus().equals(InstallmentStatus.EXPIRED) ||
+        installment.getStatus().equals(InstallmentStatus.UNPAYABLE))
+      .map(installmentMapper::mapToDto)
+      .toList();
+
+    if(installments.isEmpty())
+      throw new NotFoundException("The installment with NAV: "+nav+" was not found");
+    if(installments.size() > 1)
+      throw new ConflictErrorException("Found more than one installment processable with NAV: "+nav);
+
+    InstallmentDTO installment = calculateNewAmount(installments.getFirst(), notificationFeeCents);
+    installmentPIIRepository.save(installmentMapper.mapToModel(installment));
+
+    InstallmentUtils.setStatus(installment, installment.getStatus());
+    return installment;
+  }
+
+  private InstallmentDTO calculateNewAmount(InstallmentDTO installmentDTO, long notificationFeeCents) {
+    long oldNotificationFeeCents = installmentDTO.getNotificationFeeCents() != null ? installmentDTO.getNotificationFeeCents() : 0L;
+    long notificationFeeDifference = notificationFeeCents - oldNotificationFeeCents;
+    installmentDTO.setNotificationFeeCents(notificationFeeDifference);
```

Comment:
I don't think this is right, I think you should set the notificationFeeCents using just the new input.
The difference should be used just to update the first transfer amount